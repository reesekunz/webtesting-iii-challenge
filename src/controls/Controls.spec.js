// Test away!
import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react";

import Controls from "./Controls";

describe("<Controls />", () => {
    it("matches snapshot", () => {
      const tree = renderer.create(<Controls/>); // generates a DOM tree
  
      // snapshots are a JSON representation of the DOM tree
      expect(tree.toJSON()).toMatchSnapshot();
    });
  
    // Implementation details ->
  //   - what the props are doing...
  //   - if props are` invoked...

  // What we _want_ to test:
  //    - When our user clicks that button... what happens in the app?
    
    it("button is disabled when gate is locked ", () => {
        const toggleLocked = jest.fn();
   // set up component 
    const { getByText, getByTestId } = render(<Controls toggleLocked={toggleLocked}/>)

   // action 
    const button = getByText(/lock-unlock/i)
    console.log(button);
    fireEvent.click(button)

    // assert 
    expect(toggleLocked).toHaveBeenCalled();
 
    }); 
      

});