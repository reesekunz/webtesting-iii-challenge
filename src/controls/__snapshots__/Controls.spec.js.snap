// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> matches snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    data-testid="lock-unlock"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="open-close"
  >
    Close Gate
  </button>
</div>
`;
